src/main/java/uk/gov/hmcts/ccd/domain/service/validate/DefaultValidateCaseFieldsOperation.java

```diff
@@ -11,33 +11,38 @@
 import uk.gov.hmcts.ccd.domain.model.definition.CaseType;
 import uk.gov.hmcts.ccd.domain.model.std.CaseDataContent;
 import uk.gov.hmcts.ccd.domain.service.common.CaseTypeService;
+import uk.gov.hmcts.ccd.domain.service.processor.FieldProcessorService;
 import uk.gov.hmcts.ccd.endpoint.exceptions.ValidationException;
 
 @Service
 public class DefaultValidateCaseFieldsOperation implements ValidateCaseFieldsOperation {
 
     private final CaseDefinitionRepository caseDefinitionRepository;
     private final CaseTypeService caseTypeService;
+    private final FieldProcessorService fieldProcessorService;
 
     @Inject DefaultValidateCaseFieldsOperation(
         @Qualifier(CachedCaseDefinitionRepository.QUALIFIER) final CaseDefinitionRepository caseDefinitionRepository,
-        final CaseTypeService caseTypeService) {
+        final CaseTypeService caseTypeService,
+        final FieldProcessorService fieldProcessorService) {
         this.caseDefinitionRepository = caseDefinitionRepository;
         this.caseTypeService = caseTypeService;
+        this.fieldProcessorService = fieldProcessorService;
     }
 
     @Override
     public final Map<String, JsonNode> validateCaseDetails(String caseTypeId, CaseDataContent content) {
-        if (content == null || content.getEvent().getEventId() == null) {
+        if (content == null || content.getEvent() == null || content.getEventId() == null) {
             throw new ValidationException("Cannot validate case field because of event is not specified");
         }
         final CaseType caseType = caseDefinitionRepository.getCaseType(caseTypeId);
         if (caseType == null) {
-            throw new ValidationException("Cannot find case type definition for  " + caseTypeId);
+            throw new ValidationException("Cannot find case type definition for " + caseTypeId);
         }
         if (!hasEventId(caseType, content.getEventId())) {
-            throw new ValidationException("Cannot validate case field because of event" + content.getEventId() + " is not found in case type definition");
+            throw new ValidationException("Cannot validate case field because of event " + content.getEventId() + " is not found in case type definition");
         }
+        content.setData(fieldProcessorService.processData(content.getData(), caseType, content.getEventId()));
```

Comment:
I'm a bit puzzled by this choice. I would have imagined we would extend the DateTimeValidator and DateValidator to do validation of the formatted date. And if successful then convert. 
Instead we are converting to the internal format before doing the validation